test/storage/sql_table_concurrent_test.cpp

```diff
@@ -0,0 +1,471 @@
+      int32_t *integer = reinterpret_cast<int32_t *>(pr->AccessWithNullCheck(pr_map->at(catalog::col_oid_t(1007))));
+      if (*version < 7)
+        EXPECT_EQ(integer, nullptr);
+      else
+        EXPECT_EQ(*integer, base_val);
+    }
+
+    if ((!v) >= 8) {
+      EXPECT_NE(pr_map->find(catalog::col_oid_t(1008)), pr_map->end());
+      int64_t *bigint = reinterpret_cast<int64_t *>(pr->AccessWithNullCheck(pr_map->at(catalog::col_oid_t(1008))));
+      if (*version < 8)
+        EXPECT_EQ(bigint, nullptr);
+      else
+        EXPECT_EQ(*bigint, base_val << 31);
+    }
+  }
+
+  void ValidateTable(const storage::SqlTable &table) {
+    auto txn = txn_manager_.BeginTransaction();
+
+    auto row_pair = table.InitializerForProjectedColumns(*versioned_col_oids[!schema_version_], 100, schema_version_);
+    auto pci = new storage::ProjectedColumnsInitializer(std::get<0>(row_pair));
+    auto pc_map = new storage::ProjectionMap(std::get<1>(row_pair));
+    byte *buffer = common::AllocationUtil::AllocateAligned(pci->ProjectedColumnsSize());
+    auto table_iter = table.begin(schema_version_);
+
+    while (table_iter != table.end()) {
+      auto pc = pci->Initialize(buffer);
+      table.Scan(txn, &table_iter, pc, *pc_map, schema_version_);
+
+      for (uint i : {0u, pc->NumTuples() - 1u}) {
+        auto pr = pc->InterpretAsRow(i);
+        EXPECT_NE(pc_map->find(catalog::col_oid_t(1000)), pc_map->end());
+        int32_t *base_val = reinterpret_cast<int32_t *>(pr.AccessWithNullCheck(pc_map->at(catalog::col_oid_t(1000))));
+        ValidateTuple<storage::ProjectedColumns::RowView>(&pr, pc_map, schema_version_, *base_val);
+      }
+    }
+
+    delete[] buffer;
+    delete pci;
+    delete pc_map;
+
+    txn_manager_.Commit(txn, TestCallbacks::EmptyCallback, nullptr);
+  }
+
+  storage::RecordBufferSegmentPool buffer_pool_{100000, 100000};
+  transaction::TransactionManager txn_manager_ = {&buffer_pool_, true, LOGGING_DISABLED};
+
+  std::default_random_engine generator_;
+
+  catalog::table_oid_t table_oid_ = catalog::table_oid_t(42);
+  storage::layout_version_t schema_version_ = storage::layout_version_t(0);
+  storage::BlockStore block_store_{100, 100};
+  std::vector<catalog::Schema::Column> cols_;
+  std::vector<std::vector<catalog::col_oid_t> *> versioned_col_oids;
+  storage::GarbageCollector gc_;
+
+ private:
+};
+
+// Execute a large number of inserts concurrently with schema changes and validate
+// all of the data is consistent with what we expect in the end by embedding an id
+// and insertion time version into the tuple.
+// NOLINTNEXTLINE
+TEST_F(SqlTableConcurrentTests, ConcurrentInsertsWithSchemaChanges) {
+  const uint32_t num_iterations = 100;
+  const uint32_t txns_per_thread = 10;
+  const uint32_t num_threads = MultiThreadTestUtil::HardwareConcurrency();
+  common::WorkerPool thread_pool(num_threads, {});
+
+  for (uint32_t iteration = 0; iteration < num_iterations; iteration++) {
```

Comment:
Maybe you do not need to repeat the test by yourself. `googletest` already provides the `--gtest_repeat` flag and the `GTEST_REPEAT` environment variable.

Reply:
We're using the same setup that was already used for the `DataTable` tests.  The repetitions here are useful for increasing the likelihood that CI will error out on race conditions in the concurrent tests without having to make deep modifications to the CMake build scripts.